### PR TITLE
Add documentation for using MAAS in DeepOps

### PR DESCRIPTION
**Pull request by ajdecon**
We have a playbook for installing MAAS but no documentation pointing to it. The Canonical MAAS docs are also not great for someone new to the space.

This PR:

* Adds a document for how to install MAAS with DeepOps, including showing a simple example of deploying a VM
* Re-organizes the PXE documentation to better lay out the options we support directly

### CODE REVIEW

File: docs/pxe/minimal-pxe-container.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+   * Username: `ubuntu`
+   * Password: `deepops`
+
+## IPMI Command Reference
+
+```sh
+# Set to boot from disk, always
+# Dell
+chassis bootdev disk options=persistent
+# DGX
```
Comment (michael-balint) [inline docs/pxe/minimal-pxe-container.md:59]: just for dgx-1

File: docs/pxe/maas.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+If you want to, you can also rename the machine by clicking the name at the top of the page.
+
+Before we can deploy this machine, we have to configure it so that MAAS can turn it on or off (or tell MAAS that we will do this manually).
+
+1. On the page for the machine, look for the navigation bar *under* the machine's name.
+1. Click Configuration.
+1. Scroll down until you see the Power configuration setting.
+    If this has already been selected to a pre-configured value, you're good to go, because MAAS has autodetected how to control the power on this node!
+1. If it's not yet selected, click the drop-down menu and it will show several different types of power control available.
+    For a physical machine, the most common choice will be "IPMI".
```
Comment (michael-balint) [inline docs/pxe/maas.md:141]: Might be useful to add that if the configuration is "IPMI" (ex: for DGX) then the user should also enter the BMC login creds.

File: docs/pxe/maas.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+Before we can deploy this machine, we have to configure it so that MAAS can turn it on or off (or tell MAAS that we will do this manually).
+
+1. On the page for the machine, look for the navigation bar *under* the machine's name.
+1. Click Configuration.
+1. Scroll down until you see the Power configuration setting.
+    If this has already been selected to a pre-configured value, you're good to go, because MAAS has autodetected how to control the power on this node!
+1. If it's not yet selected, click the drop-down menu and it will show several different types of power control available.
+    For a physical machine, the most common choice will be "IPMI". 
+    MAAS also supports power control via several different types of hypervisors, such as VMWare.
```
Comment (michael-balint) [inline docs/pxe/maas.md:142]: Ditto - here the creds will be the user/password for vmware/esxi.

File: docs/pxe/maas.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+At this point, the machine will PXE boot from the MAAS server and begin installing the OS to the local disk.
+After the installation completes, the machine should show as "Deployed" in the MAAS web interface.
+And at this point your should be able to log in using the SSH key you configured in MAAS!
+
+## Scaling up
+
+This is a lot of work for a single machine, but most operations in MAAS can be done in batches.
+It's generally straightforward to auto-discover a large number of hosts at a time,
+then commission them and provision them by selecting them as a group in the Machines interface.
```
Comment (michael-balint) [inline docs/pxe/maas.md:172]: Might be useful to add to the "Deploying the test VM" section (or create a new section) which explains that when you wish to reset a system, it isn't necessary to run through commissioning, etc - just deploy once again with the desired OS settings.

File: docs/pxe/maas.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
+OS Provisioning with MAAS
+=========================
+
+DeepOps includes an Ansible [playbook for setting up MAAS](../../playbooks/provisioning/maas.yml),
+Canonical's [Metal as a Service](https://maas.io/) tool for provisioning the operating system on bare-metal servers.
+By default, this playbook will set up a single-node MAAS install which can then be used to provision other servers on your network.
+
+In most cases, you will use MAAS to provision the OS on multiple bare-metal servers.
+However, this is often difficult and time-consuming when first getting started.
```
Comment (dholt) [inline docs/pxe/maas.md:9]: > However, this is often difficult and time-consuming when first getting started.

maybe remove this line, we want to be encouraging ;)

File: docs/pxe/maas.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+* A network connection which you can use to log into the MAAS server. This may be the same network as the inter-node network, or it may be a separate network.
+
+![maas-topology](./maas-example-vms.png "Example network topology")
+
+In this example, we will use:
+
+* An internal (VM-only) network on which we'll use the subnet `192.168.1.0/24`
+* An external network connection on which we'll use the subnet `192.168.122.0/24`
+* `maas-vm`, a pre-installed Ubuntu 18.04 virtual machine
+    * `maas-vm` has IP `192.168.1.1` on the internal network, and `192.168.122.90`
```
Comment (dholt) [inline docs/pxe/maas.md:32]: might add "... on the external network" to the end of the line after the external IP

**Review by dholt — DISMISSED** [review]

**Review by michael-balint — APPROVED** [review]

**Pull request merged**